TransformationSystem/Client/TaskManager.py

```diff
@@ -714,13 +738,15 @@ def __generatePluginObject( self, plugin ):
   #############################################################################
 
   def getOutputData( self, paramDict, moduleLocation ):
-    moduleFactory = ModuleFactory()
-
-    moduleInstance = moduleFactory.getModule( moduleLocation, paramDict )
-    if not moduleInstance['OK']:
-      return moduleInstance
-    module = moduleInstance['Value']
-    return module.execute()
+    if not self.outputDataModule_o:
+      moduleFactory = ModuleFactory()
+
+      moduleInstance = moduleFactory.getModule( moduleLocation, None )
+      if not moduleInstance['OK']:
+        return moduleInstance
+      self.outputDataModule_o = moduleInstance['Value']
+    self.outputDataModule_o.paramDict = paramDict
+    return self.outputDataModule_o.execute()
```

Comment:
yes... in fact it uses a plugin, but it is always the same for a given agent. Therefore getting the plugin once as an object is good enough. In fact I will change the code and remove the "moduleLocation" parameter as it is always the same value that is a data member...